### PR TITLE
refactor: update `blas/ext/base/sdsapxsumpw` to follow current project conventions

### LINKED ISSUE #1522: [RFC]: refactor `blas/ext/base/sdsapxsumpw` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/sdsapxsumpw`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/sdsapxsumpw) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Jai0401**
Resolves #1522 

## Description
This pull request refactors `blas/ext/base/sdsapxsumpw` to follow current project conventions.

## Related Issues
#1152 
#788 

This pull request:

-   resolves #1522 
-  progresses #1152

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 @license Apache-2.0
 
-Copyright (c) 2020 The Stdlib Authors.
+Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/README.md]:
```suggestion
Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/README.md [hunk 2]
```diff
@@ -139,18 +132,11 @@ var v = sdsapxsumpw.ndarray( N, 5.0, x, 2, 1 );
...
 var sdsapxsumpw = require( '@stdlib/blas/ext/base/sdsapxsumpw' );
 
-var x;
-var i;
-
-x = new Float32Array( 10 );
-for ( i = 0; i < x.length; i++ ) {
-    x[ i ] = round( randu()*100.0 );
-}
+var x = filledarrayBy(10, 'float32', discreteUniform( 0, 100 ));
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/README.md]:
```suggestion
var x = filledarrayBy( 10, 'float32', discreteUniform( 0, 100 ) );
```

File: lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/benchmark/benchmark.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/benchmark/benchmark.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/benchmark/benchmark.js [hunk 4]
```diff
@@ -21,14 +21,19 @@
...
 var isnan = require( '@stdlib/math/base/assert/is-nan' );
 var pow = require( '@stdlib/math/base/special/pow' );
-var Float32Array = require( '@stdlib/array/float32' );
 var pkg = require( './../package.json' ).name;
 var sdsapxsumpw = require( './../lib/sdsapxsumpw.js' );
 
 
+// VARIABLES //
+
+var rand = uniform( -100.0, 100.0 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/benchmark/benchmark.js]:
```suggestion
var rand = uniform( -10.0, 10.0 );
```

File: lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/benchmark/benchmark.native.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/benchmark/benchmark.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/benchmark/benchmark.native.js [hunk 6]
```diff
@@ -36,6 +36,7 @@ var sdsapxsumpw = tryRequire( resolve( __dirname, './../lib/sdsapxsumpw.native.j
 var opts = {
 	'skip': ( sdsapxsumpw instanceof Error )
 };
+var rand = uniform( -100.0, 100.0 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/benchmark/benchmark.native.js]:
```suggestion
var rand = uniform( -10.0, 10.0 );
```

File: lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/benchmark/benchmark.ndarray.js [hunk 7]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/benchmark/benchmark.ndarray.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/benchmark/benchmark.ndarray.js [hunk 8]
```diff
@@ -21,14 +21,19 @@
...
 var isnan = require( '@stdlib/math/base/assert/is-nan' );
 var pow = require( '@stdlib/math/base/special/pow' );
-var Float32Array = require( '@stdlib/array/float32' );
 var pkg = require( './../package.json' ).name;
 var sdsapxsumpw = require( './../lib/ndarray.js' );
 
 
+// VARIABLES //
+
+var rand = uniform( -100.0, 100.0 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/benchmark/benchmark.ndarray.js]:
```suggestion
var rand = uniform( -10.0, 10.0 );
```

File: lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/benchmark/benchmark.ndarray.native.js [hunk 9]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/benchmark/benchmark.ndarray.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/benchmark/benchmark.ndarray.native.js [hunk 10]
```diff
@@ -36,6 +36,7 @@ var sdsapxsumpw = tryRequire( resolve( __dirname, './../lib/ndarray.native.js' )
 var opts = {
 	'skip': ( sdsapxsumpw instanceof Error )
 };
+var rand = uniform( -100.0, 100.0 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/benchmark/benchmark.ndarray.native.js]:
```suggestion
var rand = uniform( -10.0, 10.0 );
```

File: lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/examples/index.js [hunk 11]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/examples/index.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/lib/index.js [hunk 12]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/lib/index.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/lib/ndarray.js [hunk 13]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/lib/ndarray.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/lib/ndarray.native.js [hunk 14]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/lib/ndarray.native.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/lib/ndarray.native.js [hunk 15]
```diff
@@ -38,20 +39,16 @@ var addon = require( './sdsapxsumpw.native.js' );
...
 * // returns 25.0
 */
 function sdsapxsumpw( N, alpha, x, stride, offset ) {
 	var view;
-	if ( stride < 0 ) {
-		offset += (N-1) * stride;
-	}
-	view = new Float32Array( x.buffer, x.byteOffset+(x.BYTES_PER_ELEMENT*offset), x.length-offset ); // eslint-disable-line max-len
+	offset = minViewBufferIndex( N, stride, offset );
+	view = offsetView( x, offset );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/lib/ndarray.native.js:52]:
```suggestion

	offset = minViewBufferIndex( N, stride, offset );
	view = offsetView( x, offset );

```

File: lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/manifest.json [hunk 16]
```diff
@@ -1,40 +1,45 @@
...
+		"-lm"
+	   ],
+	   "libpath": [],
+	   "dependencies": [
+		 "stdlib/napi/export.h",
+		 "stdlib/napi/argv.h",
+		 "stdlib/napi/argv_int64.h",
+		 "stdlib/napi/argv_strided_float32array.h"
+	   ]
+	  }
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/manifest.json]: Incorrect indentation

File: lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/src/addon.c [hunk 17]
```diff
@@ -0,0 +1,48 @@
...
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, stride, argv, 1 );
+
+	napi_value v;
+	napi_status status = napi_create_double( env, (double)stdlib_strided_sdsapxsumpw( N, (float *)X, stride ), &v );
+	assert( status == napi_ok );
+
+	return v;
+}
+
+STDLIB_NAPI_MODULE_EXPORT_FCN( addon )
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/src/addon.c]: trailing line required.

File: lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/test/test.ndarray.js [hunk 18]
```diff
@@ -1,7 +1,7 @@
 /**
 * @license Apache-2.0
 *
-* Copyright (c) 2020 The Stdlib Authors.
+* Copyright (c) 2024 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/test/test.ndarray.js]:
```suggestion
* Copyright (c) 2020 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/test/test.ndarray.js [hunk 19]
```diff
@@ -118,7 +117,7 @@ tape( 'the function supports a `stride` parameter', function test( t ) {
 		2.0
 	]);
 
-	N = floor( x.length / 2 );
+	N = 4;
 	v = sdsapxsumpw( N, 5.0, x, 2, 0 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/test/test.ndarray.js]: Please remove N and use 4 directly

Comment (Pranavchiku) [reply lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/test/test.ndarray.js]: same comment at all places

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
Left comments, please address those.

**Comment by Jai0401** [comment]
@Pranavchiku thanks for the review, working on suggestions.

**Comment by Jai0401** [comment]
Hello @Pranavchiku, I have implemented the necessary changes as per our discussion. Please review the updates and let me know if any further adjustments are required. Thank you for your guidance and support.

File: lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/manifest.json [hunk 20]
```diff
@@ -1,40 +1,40 @@
...
+	   ],
+	   "include": [
+		"./include"
+	   ],
+	   "libraries": [
+		"-lm"
+	   ],
+	   "libpath": [],
+	   "dependencies": []
+	  }
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/manifest.json]: This file has incorrect indentation, you may setup editor config extension in VSCode.

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
Just fix indentation in `manifest.json` and this will be ready to merge.

**Comment by Jai0401** [comment]
Hello @Pranavchiku, I've made the requested changes. Please review and let me know if anything else is needed. Thanks for your help!

File: lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/README.md [hunk 21]
```diff
@@ -56,16 +56,14 @@ The function has the following parameters:
 -   **x**: input [`Float32Array`][@stdlib/array/float32].
 -   **stride**: index increment for `x`.
 
-The `N` and `stride` parameters determine which elements in `x` are accessed at runtime. For example, to access every other element in `x`,
+The `N` and `stride` parameters determine which elements in the strided array are accessed at runtime. For example, to compute the sum of every other element in `x`,
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/README.md:59]:
```suggestion
The `N` and stride parameters determine which elements in the strided array are accessed at runtime. For example, to compute the sum of every other element in `x`,
```

File: lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/docs/repl.txt [hunk 22]
```diff
@@ -93,8 +90,7 @@
 
     // Using offset parameter:
     > var x = new {{alias:@stdlib/array/float32}}( [ 1.0, -2.0, 3.0, 2.0, 5.0, -1.0 ] );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/docs/repl.txt]:
```suggestion
    > x = new {{alias:@stdlib/array/float32}}( [ 1.0, -2.0, 3.0, 2.0, 5.0, -1.0 ] );
```

File: lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/src/addon.c [hunk 23]
```diff
@@ -0,0 +1,48 @@
...
+	STDLIB_NAPI_ARGV( env, info, argv, argc, 3 );
+	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
+	STDLIB_NAPI_ARGV_INT64( env, stride, argv, 2 );
+	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, stride, argv, 1 );
+
+	napi_value v;
+	napi_status status = napi_create_double( env, (double)stdlib_strided_sdsapxsumpw( N, (float *)X, stride ), &v );
+	assert( status == napi_ok );
+
+	return v;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/src/addon.c]:
```suggestion
	STDLIB_NAPI_ARGV( env, info, argv, argc, 4 );
	STDLIB_NAPI_ARGV_INT64( env, N, argv, 0 );
	STDLIB_NAPI_ARVG_FLOAT( env, alpha, argv, 1 );
	STDLIB_NAPI_ARGV_INT64( env, stride, argv, 3 );
	STDLIB_NAPI_ARGV_STRIDED_FLOAT32ARRAY( env, X, N, stride, argv, 2 );

	napi_value v;
	napi_status status = napi_create_double( env, (double)stdlib_strided_sdsapxsumpw( N, alpha, X, stride ), &v );
	assert( status == napi_ok );

	return v;
```

This is incorrect. You also need to handle the scalar constant `alpha`.

File: lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/src/addon.c [hunk 24]
```diff
@@ -0,0 +1,48 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/blas/ext/base/sdsapxsumpw.h"
+#include "stdlib/napi/export.h"
+#include "stdlib/napi/argv.h"
+#include "stdlib/napi/argv_int64.h"
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/sdsapxsumpw/src/addon.c:22]:
```suggestion
#include "stdlib/napi/argv_int64.h"
#include "stdlib/napi/argv_float.h"
```

**Review by kgryte — CHANGES_REQUESTED** [review]
Thanks for working on this, @Jai0401. You're missing changes to the `manifest.json` file, which needs to be updated to include additional build configurations.

**Comment by kgryte** [comment]
See https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/blas/ext/base/dfill/manifest.json for an example.

**Comment by Jai0401** [comment]
@kgryte I've made the requested changes. Please review and let me know if anything else is needed. Thanks for your help!

**Review by Planeshifter — APPROVED** [review]
Thanks for the edits! Took a final look and this is now all looking good. Will merge shortly.

**Pull request merged**